platform/frontend/src/components/onboarding/onboarding-wizard.tsx

```diff
@@ -0,0 +1,351 @@
+"use client";
+
+import {
+  forwardRef,
+  useCallback,
+  useEffect,
+  useImperativeHandle,
+  useRef,
+  useState,
+} from "react";
+import { useHasFirstUserInteraction } from "@/lib/interaction.query";
+import { DynamicInteraction } from "@/lib/interaction.utils";
+import { useDetectedTools } from "@/lib/tool.query";
+import {
+  FRAMEWORK_DOCS,
+  FRAMEWORK_LABELS,
+  PROVIDER_INFO,
+} from "../../../../shared";
+import CopyButton from "../ui/copy-button";
+import OnboardingStep from "../ui/onboarding-step";
+import OptionButton from "../ui/option-button";
+
+export type OnboardingWizardHandle = {
+  next: () => void;
+  prev: () => void;
+  goto: (n: number) => void;
+  step: number;
+};
+
+export default forwardRef(function OnboardingWizard(
+  {
+    onStepChange,
+    onComplete,
+  }: {
+    onStepChange?: (step: number) => void;
+    onComplete?: () => void;
+  },
+  ref: React.Ref<OnboardingWizardHandle | null>,
+) {
+  const [step, setStep] = useState(0);
+  const [isTransitioning, setIsTransitioning] = useState(false);
+  const [framework, setFramework] = useState<"n8n" | "langchain" | "openwebui">(
+    "n8n",
+  );
+  const [provider, setProvider] = useState<"openai" | "anthropic">("openai");
```

Comment:
Those + frameworks should not be hardcoded this way. If we have existing type for it (like SupportedProvidersInput in frontend/src/lib/clients/api/types.gen.ts) let's use it. 

If sth is missing like the frameworks, I suggest to put it in shared package as a const / enum and reuse

Reply:
Yep I add type framework and forgot replacing here